pervane/serve.py

```diff
@@ -371,6 +368,80 @@ def _get_file_mode(path):
   return 'text/html'
 
 
+def _get_file_mod_time(path):
+  mod_secs = os.path.getmtime(path)
+  return datetime.datetime.fromtimestamp(mod_secs).strftime('%Y-%m-%d %H:%M:%S')
+
+
+def _get_request_param(arg_name):
+  return request.args.get(arg_name, '').strip()
+
+
+def _get_request_json(arg_name):
+  return request.json.get(arg_name, '').strip()
+
+
+def _to_real_path(parent_path, relative_path):
+  # ignores the heading '/' in relative_path
+  start = 1 if relative_path.startswith('/') else 0
+  absoluate_path = os.path.join(parent_path, relative_path[start:])
+  return os.path.realpath(absoluate_path)
+
+
+def _get_real_path(path):
+  if not path:
+    return (None, 'No path is given')
+
+  root_dir = _get_root_dir(trailing_separator=False)
+  path = _to_real_path(root_dir, path)
+
+  if path.startswith(root_dir):
+    return (path, None)
+
+  return (None, 'Not authorized')
+
+
+def _get_new_node_name(new_node_name):
```

Comment:
can you add  a docstring about the return type. Returning tuple and it's not obvious which element is what. Probably we should use namedtuple or an object. But you can ignore for that for this PR.